### PR TITLE
feat: Snowflake sink

### PR DESCRIPTION
**Pull request by abcpro1**
Introduce the Snowflake sink as a storage destination for Dozer pipeline transformations.
A snowflake table is updated at a configurable interval with the latest values from Dozer.

### CODE REVIEW

File: dozer-log/src/reader/checkpoint.rs [hunk 1]
```diff
@@ -56,6 +63,7 @@ impl CheckpointedLogReader {
             .await
             .map_err(|e| Error::DescribeStorage(endpoint.clone(), e))?
             .into_inner();
+        dozer_types::log::warn!("storage response: {storage_response:?}");
```
Comment (chubei) [inline dozer-log/src/reader/checkpoint.rs:66]: Looks like debug log?

File: dozer-sinks/snowflake/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,452 @@
+use std::{thread, time::Duration, usize};
+
+use dozer_api::shutdown::ShutdownReceiver;
+use dozer_ingestion_snowflake::{
```
Comment (abcpro1) [inline dozer-sinks/snowflake/src/lib.rs:4]: Currently uses `dozer_ingestion_snowflake` crate directly.

**Review by chubei — APPROVED** [review]

File: dozer-log/src/reader/checkpoint.rs (hunk 1 above)
Comment (abcpro1) [reply dozer-log/src/reader/checkpoint.rs:66]: Indeed.

File: dozer-sinks/snowflake/src/lib.rs (hunk 2 above)
Comment (abcpro1) [reply dozer-sinks/snowflake/src/lib.rs]: Switched to `odbc-api` crate.

**Pull request merged**